src/comp/comp.py

```diff
@@ -24,48 +30,55 @@ def __repr__(self):
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
+a = [i.name for i in humans if re.match(r'D', i.name)]
 print(a)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
+b = [i.name for i in humans if re.search(r'e$', i.name)]
 print(b)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
-c = []
+c = [i.name for i in humans if re.match(r'C|D|E|F|G', i.name)]
 print(c)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-d = []
+d = [i.age+10 for i in humans]
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-e = []
+e = [f'{i.name}-{i.age}' for i in humans]
 print(e)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-f = []
+f = [(i.name, i.age) for i in humans if 32 >= i.age >= 27]
 print(f)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names uppercase and the ages with 5 added to them.
 # The "humans" list should be unmodified.
+
+
+def embiggen(human):
+    bigger_name = human.name.upper()
+    bigger_age = human.age + 5
+    return Human(bigger_name, bigger_age)
+
+
 print("All names uppercase:")
-g = []
+g = [embiggen(i) for i in humans]
```

Comment:
Nice use of a function to extract the complexity.  However, I would recommend naming it something that other programmers could easily read and understand what its doing.  maybe like updateHuman() ?  embiggen() kind of throws me off a bit at first until I look into what the function is doing, while something like updateHuman  gives me some idea of what the function is doing before I start reading into it.

Reply:
embiggenHuman then? "update" is so... abstract?